routing/base/astar_algorithm.hpp

```diff
@@ -343,40 +374,46 @@ typename AStarAlgorithm<TGraph>::Result AStarAlgorithm<TGraph>::FindPath(
     return true;
   };
 
-  auto adjustEdgeWeight = [&](TVertexType const & vertex, TEdgeType const & edge) {
-    auto const len = edge.GetWeight();
-    auto const piV = graph.HeuristicCostEstimate(vertex, finalVertex);
-    auto const piW = graph.HeuristicCostEstimate(edge.GetTarget(), finalVertex);
-    auto const reducedLen = len + piW - piV;
+  auto const adjustEdgeWeight = [&](TVertexType const & vertexV, TEdgeType const & edge) {
+    auto const reducedWeight = fullToReducedLength(vertexV, edge.GetTarget(), edge.GetWeight());
+
+    CHECK_GREATER_OR_EQUAL(reducedWeight, -kEpsilon,
+                           ("Invariant violated:", reducedWeight, "<", -kEpsilon));
```

Comment:
"Invariant violated" is enough now, this form of CHECK should print everything else.

Reply:
done